### PR TITLE
93 quick pyqode

### PR DESCRIPTION
**Pull request by grace-harper-ibm**

### CODE REVIEW

File: qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py:39]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py:54]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py:86]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py:58]: Fix docstring formatting.

```
Returns:
   type: Description
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:47]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:61]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:63]: Fix docstring formatting.

```
Returns:
   type: Description
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:77]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:98]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:151]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:134]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:122]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:125]: Fix docstring formatting.

```
Returns:
   type: Description
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:137]: Fix docstring formatting.

```
Returns:
   type: Description
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:154]: Fix docstring formatting.

```
Returns:
   type: Description
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:170]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py:215]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py:31]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py:47]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py:57]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py:77]: Fix docstring formatting.

```
Returns:
   type: Description
```

File: qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/qlibrary_display/proxy_model_qlibrary.py:74]: Fix docstring formatting.

```
Arguments:
   arg_name (type): Description
   arg_name (type): Description
   ...
```

File: qiskit_metal/designs/design_base.py
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/designs/design_base.py:345]: Please keep this,  Allows user to obtain a QComponent by interger id, vs string id.

**Review by JeremyDrysdale-ibm — CHANGES_REQUESTED** [review]
What is the purpose of these empty files?
- qiskit_metal/_gui/widgets/developer_mode/DeveloperMode.py
- qiskit_metal/_gui/widgets/developer_mode/__init__.py

File: qiskit_metal/designs/design_base.py
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/designs/design_base.py:436]: The intentions is to add 100,000+ qcomponents in a qdesign.  Why is this code doing this?

Comment (grace-harper-ibm) [reply qiskit_metal/designs/design_base.py:436]: 2 reasons.
The first is that this mode is for developing an individual QComponent so we don't anticipate anyone having more than 1-5 QComponents in their circuit.

The second reason is that this is part of the code while will get overhauled in the next PR.  Because we're no longer sending individual filenames to the design_base, we actually need a clean way for design_base to read the `dirtied_files` from the File Watcher. 
Currently the File Watcher is part of the Model for the QLibrary display, but in the next PR, that is getting pulled out and made into a separate piece, at which point I'll reimplement the `design_base` logic to re-import any edited file instead of just the QComponents being used.

File: qiskit_metal/designs/design_base.py
Comment (grace-harper-ibm) [reply qiskit_metal/designs/design_base.py:345]: Okay. 
Is there any issue with me deleting the comment instructing it to be deleted?

Comment (priti-ashvin-shah-ibm) [reply qiskit_metal/designs/design_base.py:345]: No, just want to keep the method.. Thanks.

Comment (priti-ashvin-shah-ibm) [reply qiskit_metal/designs/design_base.py:345]: The method has been commented out.  Please put it back in as active code.

**Review by marcolincs — APPROVED** [review]
This is good. GUI developed according to plan.
- Code changes are nice and clean
- functionally working as by design.

NOTE: need a follow-up after approval of this PR. Indeed, this solution is designed on the assumptions that users will always put their components inside the qlibrary folder. However that will not be advisable once we distribute qiskit-metal with PyPI or conda since it will be installed in the env-specific site-packages folder. Additionally users might have a separate change-control system (git or other) for their qcomponents that require them to be in a very different location. We might need to remove the qlibrary-location limitation next. @ThomasGM4 , @priti-ashvin-shah-ibm , thoughts?

**Comment by priti-ashvin-shah-ibm** [comment]
> This is good. GUI developed according to plan.
> 
>     * Code changes are nice and clean
> 
>     * functionally working as by design.
> 
> 
> NOTE: need a follow-up after approval of this PR. Indeed, this solution is designed on the assumptions that users will always put their components inside the qlibrary folder. However that will not be advisable once we distribute qiskit-metal with PyPI or conda since it will be installed in the env-specific site-packages folder. Additionally users might have a separate change-control system (git or other) for their qcomponents that require them to be in a very different location. We might need to remove the qlibrary-location limitation next. @ThomasGM4 , @priti-ashvin-shah-ibm , thoughts?

@marcolincs I think we are all on the same page.  I believe that this pr is just to stop using pyqode.  One step at a time.

In design_base and base.py, my understanding is there is no assumption, as to where the QComponents are placed.  I believe @grace-harper-ibm plans to, in future prs,  address more details later. @grace-harper-ibm , can you comment on this?

**Review by JeremyDrysdale-ibm — APPROVED** [review]

**Pull request merged**
